python/cudf/cudf/core/column/string.py

```diff
@@ -1786,6 +1805,76 @@ def translate(self, table, **kwargs):
             cpp_translate(self._column, table), **kwargs
         )
 
+    def normalize_spaces(self, **kwargs):
+        return self._return_or_inplace(
+            cpp_normalize_spaces(self._column), **kwargs
+        )
+
+    def tokenize(self, delimiter=" ", **kwargs):
```

Comment:
The default delimiter for the 'token*' methods should not be space.
https://github.com/rapidsai/cudf/blob/aedcbe875c2b370951f489889764bcbde7f308b4/python/nvstrings/nvtext.py#L7

Reply:
@davidwendt `tokenize.cu` requires a valid string scalar. Can you elaborate?

Reply:
@davidwendt `tokenize.cu` requires a valid `string_scalar`. Can you elaborate?

Reply:
Maybe another oversight. You can make an empty string "" as the default parameter which would match the libcudf signature.
https://github.com/rapidsai/cudf/blob/0242d70fe2f4870a5e28e27361d4fe2215fbcd7c/cpp/include/nvtext/tokenize.hpp#L55

Reply:
@davidwendt it looks like the legacy implementation assumes any `c < ' '` is whitespace. If we want to retain that functionality, we'll need to allow `tokenize` and `count_tokens` to accept an invalid scalar. It looks like this is already implemented in `cudf/strings`, we're just circumventing it by supplying `cudf::string_scalar{""}` as the default arguments.

Reply:
The whitespace functionality is available with `cudf::string_scalar("")`  -- empty string.
https://github.com/rapidsai/cudf/blob/0242d70fe2f4870a5e28e27361d4fe2215fbcd7c/cpp/include/nvtext/tokenize.hpp#L50
But we could also support an invalid scalar for the same thing if you think that would be better.

Reply:
Oh, interesting. I misread `string_scalar{""}` as `string_scalar{" "}`.